yt/frontends/chimera/io.py

```diff
@@ -0,0 +1,306 @@
+                        "num_lumin_2",
+                        "num_lumin_3",
+                        "num_lumin_4",
+                    ):
+                        dims = f["mesh"]["array_dimensions"]
+                        n_groups = f["radiation"]["raddim"][0]
+                        n_hyperslabs = f["mesh"]["nz_hyperslabs"][()]
+                        ergmev = 1.602177e-6
+                        cvel = 2.99792458e10
+                        h = 4.13567e-21
+                        ecoef = 4.0 * np.pi * ergmev / (h * cvel) ** 3
+                        radius = f["mesh"]["x_ef"][()]
+                        agr_e = f["fluid"]["agr_e"][()]
+                        cell_area_GRcorrected = 4 * np.pi * radius ** 2 / agr_e ** 4
+                        psi1_e = f["radiation"]["psi1_e"]
+                        energy_edge = f["radiation"]["unubi"][()]
+                        energy_center = f["radiation"]["unui"][()]
+                        d_energy = []
+                        for i in range(0, n_groups):
+                            d_energy.append(energy_edge[i + 1] - energy_edge[i])
+                        d_energy = np.array(d_energy)
+                        e2de = energy_center ** 2 * d_energy
+                        e3de = energy_center ** 3 * d_energy
+                        e5de = energy_center ** 5 * d_energy
+                        species = int(fname[-1]) - 1
+                        if fname in ("lumin_1", "lumin_2", "lumin_3", "lumin_4"):
+                            lumin = (
+                                np.sum(psi1_e[:, :, :, species] * e3de, axis=3)
+                                * np.tile(
+                                    cell_area_GRcorrected[1 : dims[0] + 1],
+                                    (int(dims[2] / n_hyperslabs), dims[1], 1),
+                                )
+                                * (cvel * ecoef * 1e-51)
+                            )
+                        elif fname in (
+                            "num_lumin_1",
+                            "num_lumin_2",
+                            "num_lumin_3",
+                            "num_lumin_4",
+                        ):
+                            lumin = (
+                                np.sum(psi1_e[:, :, :, species] * e2de, axis=3)
+                                * np.tile(
+                                    cell_area_GRcorrected[1 : dims[0] + 1],
+                                    (int(dims[2] / n_hyperslabs), dims[1], 1),
+                                )
+                                * (cvel * ecoef * 1e-51)
+                            )
+                        dat_1 = lumin[:, :, :].transpose()
+
+                    elif fname == "shock":
+                        sys.exit(
+                            "Error: Currently disfuctional, please select another option"
+                        )  # This overlay does not currently work, calculation code is left here for future refinement and implementation.
```

Comment:
I believe this is a left over from an older version of the Skeleton frontend, and in any case this isn't the right way to raise an error, please check this.